"long" types need to be listed as dependency - yarn v2

### Problem description
Need to list `@types/long` as dependency

### Reproduction steps
Typescript build fails when using Yarn v2+ in pnp mode

### Environment
 - OS name, version and architecture: Arch Linux
 - Node version 14.18.2
 - Node installation method: nvm
 - Package name and version: @grpc/grpc-js@1.5.0

### Additional context
Building with yarn and Typescript gives:

```
error TS2688: Cannot find type definition file for 'long'.

/// <reference types="long" />
```

I solved it by adding this to my `.yarnrc.yml`, which I believe means `@types/long` needs to be added as an explicit dependency in this package

```
packageExtensions:
  "@grpc/grpc-js@*":
    dependencies:
      "@types/long": "*"
```

Well, grpc-js references "long" in 
build\src\generated\google\protobuf\Int64Value.d.ts
and more than 10 other files. yarn 2 or 3 report this as
```
.yarn/cache/@grpc-grpc-js-npm-1.5.3-4edae7a246-19d75ac09e.zip/node_modules/@grpc/grpc-js/build/src/generated/google/protobuf/Int64Value.d.ts:1:23 - error TS2688: Cannot find type definition file for 'long'.

1 /// <reference types="long" />
                        ~~~~
```

This is not related to yarn pnp.

With nodeLinker node_modules @types/long is pulled in via
```
>yarn why @types/long
├─ @grpc/proto-loader@npm:0.6.9
│  └─ @types/long@npm:4.0.1 (via npm:^4.0.1)
│
└─ protobufjs@npm:6.11.2
   └─ @types/long@npm:4.0.1 (via npm:^4.0.1)
```
so it doesn't matter, that grpc-js doesn't.

--- a/packages/grpc-js/tsconfig.json
+++ b/packages/grpc-js/tsconfig.json
@@ -6,7 +6,8 @@
     "target": "es2017",
     "module": "commonjs",
     "resolveJsonModule": true,
-    "incremental": true
+    "incremental": true,
+    "types": ["mocha"]
   },
   "include": [
     "src/**/*.ts",

